examples/solids/index.rst

```diff
@@ -0,0 +1,348 @@
+.. _example-petsc-elasticity:
+
+Solid mechanics elasticity mini-app
+========================================
+
+This example is located in the subdirectory :file:`examples/solids`. It solves
+the steady-state static balance momentum equations using unstructured high-order
+finite element/spectral element spatial discretizations. Moreover, the solid
+mechanics elasticity example has been developed using PETSc, so that the
+pointwise physics (defined at quadrature points) is separated from the
+parallelization and meshing concerns.
+
+
+.. _problem-linear-elasticity:
+
+----------------------------------------
+
+The strong form of the static balance linear momentum at small strain for the three-dimensional linear elasticity problem is given by :cite:`hughes2012finite`:
+
+.. math::
+   :label: lin-elas
+   
+   \nabla \cdot \boldsymbol{\sigma} + \boldsymbol{g} = \boldsymbol{0} 
+
+
+where :math:`\boldsymbol{\sigma}`, :math:`\boldsymbol{\epsilon}`
+and :math:`\boldsymbol{g}` are stress, strain, and forcing functions,
+respectively. Integrating by parts on the divergence term, we arrive at the weak form the of equation :math:numref:`lin-elas`:
+
+.. math::
+
+   \int_{\Omega}{ \nabla \boldsymbol{v} \colon \boldsymbol{\sigma}} dV - \int_{d\Omega}{\boldsymbol{v} \cdot \left(\boldsymbol{\sigma}_t \cdot \hat{\boldsymbol{n}}\right)} dS + \int_{\Omega}{\boldsymbol{v} \cdot \boldsymbol{g}} dV = 0
+
+where :math:`\boldsymbol{\sigma}_t \cdot \hat{\boldsymbol{n}}` is typically
+replaced with a boundary condition.
+
+The stress-strain relationship (constitutive law) is given by:
+
+.. math::
+   :label: linear-stress-strain
+
+   \boldsymbol{\sigma} = \mathsf{C} \boldsymbol{\epsilon},
+
+where 
+
+.. math::
+
+   \boldsymbol{\epsilon} = \dfrac{1}{2}\left(\nabla \boldsymbol{u} + \nabla \boldsymbol{u}^T \right)
+
+is the (symmetric) strain tensor.
+For notational convenience, we suppose the symmetric second order tensors :math:`\bm \sigma` and :math:`\bm \epsilon` are expressed as vectors of length 6 using `Voigt notation <https://en.wikipedia.org/wiki/Voigt_notation>`_, in which case the fourth order material stiffness tensor :math:`\mathsf C` can be represented as a :math:`6\times 6` symmetric matrix
```

Comment:
FWIW maybe material stiffness tensor could be confused by someone for the element stiffness matrix? Hughes and other classical authors call this "material moduli matrix" or "elasticity matrix of moduli".

Reply:
Ogden and Wriggers both use "elasticity tensor". "Stiffness" is used sometimes to distinguish from "compliance", though the latter is much less common. People also find this usage sometimes: https://en.wikipedia.org/wiki/Hooke%27s_law#Matrix_representation_.28stiffness_tensor.29

I don't think confusion with a finite element stiffness matrix is problematic. I don't like using "matrix" for the mathematical object ("tensor" is better). No strong feelings between "elasticity tensor", "material tensor", "elastic moduli tensor", or the wording in the PR.